testing/sample_framework/src/android/android_app_framework.cc

```diff
@@ -478,6 +479,40 @@ std::string ReadTextInput(const char* title, const char* message,
   return g_text_entry_field_data->ReadText(title, message, placeholder);
 }
 
+void SetEnvironmentVariableFromExtra(const char* extra_name, JNIEnv* env,
+                                     jobject intent,
+                                     jmethodID get_string_extra) {
+  jstring extra_value_jstring = (jstring)env->CallObjectMethod(
+      intent, get_string_extra, env->NewStringUTF(extra_name));
+
+  if (extra_value_jstring != nullptr) {
+    const char* extra_value =
+        env->GetStringUTFChars(extra_value_jstring, nullptr);
+    setenv(extra_name, extra_value == nullptr ? "" : extra_value,
+           /*overwrite=*/1);
+    env->ReleaseStringUTFChars(extra_value_jstring, extra_value);
+  }
+}
+
+void SetExtrasAsEnvironmentVariables() {
+  JNIEnv* env = app_framework::GetJniEnv();
+  jobject activity = g_app_state->activity->clazz;
+
+  jclass activity_clazz = env->GetObjectClass(activity);
+  jmethodID get_intent = env->GetMethodID(activity_clazz, "getIntent",
+                                          "()Landroid/content/Intent;");
+  jobject intent = env->CallObjectMethod(activity, get_intent);
```

Comment:
Same here, I think you should be calling DeleteLocalRef at the end of the function.

Reply:
Done.